packages/lesswrong/server/sqlConnection.ts

```diff
@@ -113,7 +120,13 @@ export const createSqlConnection = async (url?: string): Promise<SqlClient> => {
   });
 
   try {
-    await Promise.all(onConnectQueries.map((query) => db.any(query)));
+    await Promise.all(onConnectQueries.map((query) => {
+      return db.tx(async (transaction) => {
+        // Set advisory lock to ensure only one server runs each query at a time
+        await transaction.any(`SELECT pg_advisory_xact_lock(${getLockKey(query)});`);
```

Comment:
Just to be safe, we should probably set a timeout with `SET LOCAL lock_timeout = '5s'` (or however long we're willing to wait).